### PR TITLE
Add NotInTransactionException

### PR DESCRIPTION
**Pull request by dain**
Code is more readable with a typed exception.

### CODE REVIEW

File: presto-main/src/main/java/io/prestosql/transaction/InMemoryTransactionManager.java [hunk 1]
```diff
@@ -270,16 +269,11 @@ private TransactionMetadata getTransactionMetadata(TransactionId transactionId)
     {
         TransactionMetadata transactionMetadata = transactions.remove(transactionId);
         if (transactionMetadata == null) {
-            return immediateFailedFuture(unknownTransactionError(transactionId));
+            return immediateFailedFuture(new NotInTransactionException());
```
Comment (findepi) [inline presto-main/src/main/java/io/prestosql/transaction/InMemoryTransactionManager.java]: Previously you'd include `transactionId` in the message

File: presto-main/src/test/java/io/prestosql/execution/TestCommitTask.java [hunk 2]
```diff
@@ -112,8 +112,7 @@ public void testUnknownTransactionCommit()
             getFutureValue(new CommitTask().execute(new Commit(), transactionManager, metadata, new AllowAllAccessControl(), stateMachine, emptyList()));
             fail();
         }
-        catch (PrestoException e) {
-            assertEquals(e.getErrorCode(), UNKNOWN_TRANSACTION.toErrorCode());
+        catch (NotInTransactionException expected) {
```
Comment (findepi) [inline presto-main/src/test/java/io/prestosql/execution/TestCommitTask.java]: Should `NotInTransactionException` be a `PrestoException` with `UNKNOWN_TRANSACTION`? 
Or `UNKNOWN_TRANSACTION` should be removed? (it's unused now)

**Comment by dain** [comment]
@findepi I applied all of your comments

**Comment by dain** [comment]
> No objections. I don't see the purpose yet (because there is no catch except for the tests).

`Session.getRequiredTransactionId` now throws a Presto exception instead of a illegal state exception, which I ran into while debugging something.

**Pull request merged**